src/main/java/com/Repository/AssociationRepository.java

```diff
@@ -1,12 +1,16 @@
 package com.Repository;
 
+import com.Model.Event;
 import com.Model.PatientEventAssociation;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
-import org.springframework.stereotype.Component;
-
-import java.util.List;
 
 public interface AssociationRepository extends CrudRepository<PatientEventAssociation, Long> {
 
+    @Query("SELECT DISTINCT p.event FROM PatientEventAssociation p " +
```

Comment:
Wonder if it's possible to achieve this just by naming the method correctly (i.e. without having to write the SQL query explicitly). Nice little exercise!